### PR TITLE
Updated dependency versions and Sprotty protocol type definitions, added ci workflow

### LINKED ISSUE #86: Update Sprotty server (latest protocol version, dependencies, Gradle build)
In our Java project (based on OSGi and Eclipse) we would like to use Sprotty server as backend. To achieve this we need several updates to the server:
- The server is outdated and not built on latest protocol version.
- The Gradle build does not run (OSGi manifest headers would be still required).
- The dependencies are outdated.

### PR DESCRIPTION
**Pull request by sailingKieler**
Fixes #86 based on the changes in #85 of @NiklasRentzCAU.

### CODE REVIEW

File: .github/workflows/build.yml [hunk 1]
```diff
@@ -0,0 +1,24 @@
...
+        os: [ubuntu-latest, windows-latest, macos-latest]
+    steps:
+    - uses: actions/checkout@v4
+    - name: Set up JDK 11 ☕
+      uses: actions/setup-java@v4
+      with:
+        java-version: 11
+        distribution: 'temurin'
+    - name: Setup Gradle 🛠️
+      uses: gradle/gradle-build-action@v3
```
Comment (spoenemann) [inline .github/workflows/build.yml:18]: Do we need this if we use the Gradle wrapper?

File: .github/workflows/build.yml [hunk 2]
```diff
@@ -0,0 +1,24 @@
...
+    - name: Use Node.js
+      uses: actions/setup-node@v4
+      with:
+        node-version: '20'
+    - name: Build with Gradle 🏗️
+      run: ./gradlew build test npmInstall
```
Comment (spoenemann) [inline .github/workflows/build.yml:24]: I think `test` is included in `build`.

**Review by spoenemann — APPROVED** [review]
Great, thanks! Just to minor remarks about the build

**Comment by spoenemann** [comment]
Two more things I noticed after running the build:
- There's an untracked package-lock.json in examples, we should add that to git
- There are a lot of generated Xtend.prefs and Java.prefs files, shall we put them to .gitignore?

File: .github/workflows/build.yml (hunk 1 above)
Comment (sailingKieler) [reply .github/workflows/build.yml:18]: It cares about caching the actual gradle installation, otherwise it need to be downloaded each and every time.

**Comment by sailingKieler** [comment]
> Two more things I noticed after running the build:
> 
> * There's an untracked package-lock.json in examples, we should add that to git
Sure. I just kept as it was before.

> * There are a lot of generated Xtend.prefs and Java.prefs files, shall we put them to .gitignore?
I didn't pay attention to them. Yes, we can ignore them.

**Pull request merged**